federation-js/src/composition/validate/postComposition/__tests__/executableDirectivesIdentical.test.ts

```diff
@@ -81,6 +82,12 @@ describe('executableDirectivesIdentical', () => {
       Array [
         Object {
           "code": "EXECUTABLE_DIRECTIVES_IDENTICAL",
+          "locations": Array [
+            Object {
+              "column": 9,
+              "line": 2,
+            },
+          ],
```

Comment:
What's the expected use case for post composition error locations? Are we going to be able to represent these in studio?

From strictly a typical gateway use case, the location is not actionable to me since the supergraph is more or less hidden away from me as an implementation detail of the gateway.

Reply:
I might not be understanding when post composition error rules are used. Are the composition errors we get on the frontend only the pre and during composition errors? Maybe worth nothing that the supergraph is visible to users in studio.

Reply:
Good to know - I think we'll find these useful then (not that I would've had you remove them or anything, in any case I think it's good they're there).

Precomposition locations will apply to subgraph SDL, and postcomposition locations will apply to supergraph SDL. The biggest difference there is that the supergraph SDL is compiled, so it only sorta makes sense to point to a spot on it and say "this is wrong, fix it", since nobody _authored_ the supergraph SDL.

And on second thought, in the case of errors there won't _be_ a supergraph SDL. I'm not sure we'll be able to utilize these until we have a concept of warnings (which would give us both a supergraph SDL and a set of errors to address). That is currently being talked about though.

Reply:
Yes that makes sense to me. composition error hints like “hey fix this on line x” will only apply to the subgraph asks.